core/secrets/secret.go

```diff
@@ -77,16 +77,21 @@ func ParseURI(str string) (*URI, error) {
 	if idStr == "" {
 		idStr = u.Opaque
 	}
-	matches := secretURIParse.FindStringSubmatch(idStr)
-	if matches == nil {
+	valid := secretURIParse.MatchString(idStr)
+	if !valid {
```

Comment:
Do we need to consider backward compatibility？

Reply:
The regexp handles both cases, with and without a model uuid. The unit tests confirm it all works.